lib/rubygems/package/tar_header.rb

```diff
@@ -127,7 +127,8 @@ def self.from(stream)
   end
 
   def self.strict_oct(str)
-    return str.strip.oct if /\A[0-7]*\z/.match?(str.strip)
+    str.strip!
```

Comment:
I don't feel strongly about this, but I'd rather see this public class method replaced with a strict_octal! method that modifies its argument than simply changed to do so. Better to totally break someone using this method by renaming it than subtly break things by mutating the argument.